publisher/publisher.go

```diff
@@ -230,11 +232,21 @@ func downloadFile(url, destPath string) error {
 
 func downloadArtifacts(conf config, schema uploadArtifactsSchema) error {
 	for _, artifactSchema := range schema {
+		// sanitize optional args as nil values will skip download
+		if artifactSchema.OS == nil {
+			artifactSchema.OS = []string{""}
+		}
+		if artifactSchema.SysManager == nil {
+			artifactSchema.SysManager = []string{""}
+		}
+
 		for _, arch := range artifactSchema.Arch {
-			for _, sysMngr := range artifactSchema.SysManager {
-				err := downloadArtifact(conf, artifactSchema.Src, arch, sysMngr)
-				if err != nil {
-					return err
+			for _, os := range artifactSchema.OS {
+				for _, sysMngr := range artifactSchema.SysManager {
+					err := downloadArtifact(conf, artifactSchema.Src, arch, os, sysMngr)
```

Comment:
This works for the download only, but the upload will be as before and therefore uploading only on artifact